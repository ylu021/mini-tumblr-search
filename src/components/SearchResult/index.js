import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

const SearchResult = ({posts, fetching}) => {
  if (fetching) {
    return <p>Loading</p>;
  }

  if (!posts) {
    return <p />;
  }

  return (
    <div>
      <h1>Image result from Tumblr</h1>
      <ul>
        {
          posts.map(({id, image, url}) =>
            <li key={id}>
              <img src={image} />
              <a
                href={url}
                target='_blank'
              >
                {'source'}
              </a>
            </li>
          )
        }
      </ul>
    </div>
  );
};

SearchResult.propTypes = {
  fetching: PropTypes.bool.isRequired,
  posts: PropTypes.array
};

SearchResult.defaultProps = {
  posts: null
};

const mapStateToProps = (state) => ({
  fetching: state.posts.fetching,
  posts: state.posts.posts
});

export {SearchResult as SearchResultDummy};
export default connect(mapStateToProps)(SearchResult);
