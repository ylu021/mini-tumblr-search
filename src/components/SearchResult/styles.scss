.info {
  color: tomato;
  font-weight: 600;
  font-size: large;
}

ul {
  list-style: none;
}

a {
  text-decoration: none;
}

.resultcontainer {
  display: flex;
  flex-flow: row;
  flex-wrap: wrap;
  width: 100%;
  padding: 0;
  margin: 0;
  justify-content: center;
}

.img {
  width: 100%;
  height: inherit;
  transition: all 200ms ease-in;
  // stylelint-disable 
  &:hover {
    width: 100%;
    height: auto;
    transition: all 200ms ease-in;
    box-shadow: 0 0 1.5rem black;
  }
}

.resultitem {
  margin-top: 2rem;
  margin-right: 2.5rem;
}

.imgcontainer {
  width: 250px;
  height: 250px;
  overflow: hidden;
  border-radius: 0.5rem;
  position: relative;
  transition: all 200ms ease-in;

  &:hover {
    z-index: 2;
    transition: all 200ms ease-in;
    transform: scale(1.5);
  }
}
