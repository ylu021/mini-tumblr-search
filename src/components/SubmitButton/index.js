import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {requestingPosts} from '../../actions/posts';
import styles from './styles.scss';

const SubmitButton = ({dispatch, fetching, query}) => {
  const handleSubmit = () => dispatch(requestingPosts(query));

  return (
    <button
      className={styles.submitbutton}
      disabled={fetching || !query}
      onClick={!fetching && handleSubmit}
    >
      Search
    </button>
  );
};

SubmitButton.propTypes = {
  dispatch: PropTypes.func.isRequired,
  fetching: PropTypes.bool.isRequired,
  query: PropTypes.string
};

SubmitButton.defaultProps = {
  query: null
};

const mapStateToProps = (state) => ({
  fetching: state.posts.fetching
});

export {SubmitButton as SubmitButtonDummy};
export default connect(mapStateToProps)(SubmitButton);
