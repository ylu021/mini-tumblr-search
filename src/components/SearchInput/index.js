import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {requestingPosts} from '../../actions/posts';
import SubmitButton from '../SubmitButton';

class SearchInput extends Component {
  state = {
    query: null
  };

  constructor (props) {
    super(props);
  }

  render () {
    const {query} = this.state;

    return (
      <form>
        <input
          name={'query'}
          onChange={
            (e) => this.setState({
              query: e.target.value
            })
          }
          type={'text'}
        />
        <SubmitButton query={query} />
      </form>
    );
  }
}

export {SearchInput as SearchInputDummy};
export default SearchInput;
