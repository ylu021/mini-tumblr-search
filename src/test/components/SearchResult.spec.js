import React, {Component} from 'react';
import {SearchResultDummy as SearchResult} from '../../components/SearchResult';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

const mockResult = [
  {
    id: 1,
    image: 'http://image.jpg',
    url: 'http://tumblritem.com/item/1'
  },
  {
    id: 2,
    image: 'http://image2.jpg',
    url: 'http://tumblritem.com/item/2'
  }
];

describe('<SearchResult />', () => {
  it('should correctly renders loading at fetching state', () => {
    const tree = renderer.create(<SearchResult fetching={true} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should correctly renders null when no posts are find', () => {
    const tree = renderer.create(<SearchResult fetching={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should correctly renders the posts', () => {
    const tree = renderer.create(<SearchResult fetching={false} posts={mockResult} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
