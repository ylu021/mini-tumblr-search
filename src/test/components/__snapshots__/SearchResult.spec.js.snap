// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> should correctly renders loading at fetching state 1`] = `
<p>
  Loading
</p>
`;

exports[`<SearchResult /> should correctly renders null when no posts are find 1`] = `<p />`;

exports[`<SearchResult /> should correctly renders the posts 1`] = `
<div>
  <h1>
    Image result from Tumblr
  </h1>
  <ul>
    <li>
      <img
        src="http://image.jpg"
      />
      <a
        href="http://tumblritem.com/item/1"
        target="_blank"
      >
        source
      </a>
    </li>
    <li>
      <img
        src="http://image2.jpg"
      />
      <a
        href="http://tumblritem.com/item/2"
        target="_blank"
      >
        source
      </a>
    </li>
  </ul>
</div>
`;
